File: #13 Roman to Integer/13.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        ans = self.roman(s[0])
        for i in range(1, len(s)):
            if self.roman(s[i]) > self.roman(s[i-1]):
                ans += self.roman(s[i]) - 2*self.roman(s[i-1])
            else: ans += self.roman(s[i])
        return ans
    
    def roman(self, c):
        if c == 'I': return 1
        elif c == 'V': return 5
        elif c == 'X': return 10
        elif c == 'L': return 50
        elif c == 'C': return 100
        elif c == 'D': return 500
        elif c == 'M': return 1000
        else: return 0
        

File: 0013. Roman to Integer/13.cpp
class Solution {
public:
    int roman(char c){
        switch(c){
            case 'I': return 1;  
            case 'V': return 5;  
            case 'X': return 10;  
            case 'L': return 50;  
            case 'C': return 100;  
            case 'D': return 500;  
            case 'M': return 1000;  
            default: return 0;  
        }
    }

    int romanToInt(string s) {
        int ans = roman(s[0]);
        for(int i=1; i<s.length(); ++i){
            if(roman(s[i]) > roman(s[i-1]))
                ans += (roman(s[i]) - 2 * roman(s[i-1]));
            else
                ans += roman(s[i]);
        }
        return ans;
    }
};